src/views/search/components/SearchBar.tsx

```diff
@@ -0,0 +1,137 @@
+
+import { useNavigateToSearchResults } from '../hooks/useNavigateToSearchResults';
+import { AdvancedSearchInputs } from '../utils/types';
+
+import AdvancedSearchModal from './AdvancedSearchModal/AdvancedSearchModal';
+import SearchSuggestBox, { SearchSuggestResult } from './SearchSuggestBox';
+
+import './search-bar.scss';
+
+const SearchBar: FC = () => {
+  const { t } = useKubevirtTranslation();
+  const { createModal } = useModal();
+
+  const [searchQuery, setSearchQuery] = useState('');
+  const [isSearchSuggestBoxOpen, setIsSearchSuggestBoxOpen] = useState(false);
+  const [isSearchInProgress, setIsSearchInProgress] = useState(false);
+  const [searchSuggestResult, setSearchSuggestResult] = useState<SearchSuggestResult>();
+
+  const searchInputRef = useRef<HTMLInputElement>();
+  const searchSuggestBoxRef = useRef<HTMLDivElement>();
+
+  const [vmSuggestions, vmSuggestionsLoaded] = useVirtualMachineSearchSuggestions(searchQuery);
+  const navigateToSearchResults = useNavigateToSearchResults();
+
+  useEffect(() => {
+    if (searchQuery) {
+      setIsSearchSuggestBoxOpen(true);
+      setIsSearchInProgress(!vmSuggestionsLoaded);
+      setSearchSuggestResult(vmSuggestions);
+    } else {
+      setIsSearchSuggestBoxOpen(false);
+      setIsSearchInProgress(false);
+      setSearchSuggestResult(undefined);
+    }
+  }, [searchQuery, vmSuggestions, vmSuggestionsLoaded]);
+
+  const onSearchInputChange = debounce<SearchInputProps['onChange']>((_, value) => {
+    setSearchQuery(value);
+  }, 500);
+
+  const onSearchInputClear = () => {
+    setSearchQuery('');
+  };
+
+  const showSearchModal = useCallback(
+    (prefillInputs?: AdvancedSearchInputs) => {
+      createModal(({ isOpen, onClose }) => (
+        <AdvancedSearchModal
+          onSubmit={(searchInputs) => {
+            navigateToSearchResults(searchInputs);
+            onClose();
+          }}
+          isOpen={isOpen}
+          onClose={onClose}
+          prefillInputs={prefillInputs}
+        />
+      ));
+    },
+    [createModal, navigateToSearchResults],
+  );
+
+  const searchInput = (
+    <SearchInput
+      id="vm-search-input"
+      onChange={onSearchInputChange}
+      onClear={onSearchInputClear}
+      onClick={() => searchQuery && setIsSearchSuggestBoxOpen(true)}
+      placeholder={t('Search VirtualMachines')}
+      ref={searchInputRef}
+      value={searchQuery}
+    />
+  );
+
+  const searchSuggestBox = (
+    <Menu
+      aria-label={t('Search suggest box')}
+      className="pf-v6-u-py-0"
+      ref={searchSuggestBoxRef}
+      role="dialog"
+    >
+      <SearchSuggestBox
+        isSearchInProgress={isSearchInProgress}
+        navigateToSearchResults={navigateToSearchResults}
+        searchQuery={searchQuery}
+        searchSuggestResult={searchSuggestResult}
+        showSearchModal={showSearchModal}
+      />
+    </Menu>
+  );
```

Comment:
use in place or extraxct it, no need a nested component

Reply:
There is nothing wrong with extracting a logical JSX element into an appropriately named variable that is used just once.

Using it in place (inlining) leads to bigger JSX chunks which are less readable. Extracting it into a separate module adds more overhead and complexity compared to just having a variable.